src/Operators/spectralelement.jl

```diff
@@ -1650,3 +1665,36 @@ function rmatmul2(W, S, i, j)
     end
     return r
 end
+
+
+
+function apply_operator(
+    op::Union{
+        Divergence{()},
+        WeakDivergence{()},
+        Gradient{()},
+        WeakGradient{()},
+        Curl{()},
+        WeakCurl{()},
+    },
+    space,
+    slabidx,
+    arg,
+)
+    FT = Spaces.undertype(space)
+    RT = operator_return_eltype(op, eltype(arg))
+    zero_value = zero(RT)
+
+    if slab_type(space) === DataLayouts.IJF
+        QS = Spaces.quadrature_style(space)
+        Nq = Quadratures.degrees_of_freedom(QS)
+        out = DataLayouts.IJF{RT, Nq}(MArray, FT)
+        fill!(parent(out), zero(FT))
+        out[slab_index(1, 1)] = zero_value
+    else
+        out = DataLayouts.IF{RT, 1}(MArray, FT)
+        out[slab_index(1)] = zero_value
+    end
+
+    return Field(SArray(out), space)
+end
```

Comment:
```suggestion
function apply_operator(op::SpectralElementOperator{()}, space, _, arg)
    RT = operator_return_eltype(op, eltype(arg))
    return map(Returns(zero(RT)), space)
end
```
This can be simplified, especially if you make the dimensions a parameter of the abstract `SpectralElementOperator` type. If you don't want to change the abstract type you can also keep the current `Union{...}` type.